from os.path import join

SAMPLES = ["SRR4238351", "SRR4238355", "SRR4238359", "SRR4238379"]
BASE_DIR = '/home/sebastien/Bureau/toy_examples'
GENOME_DIR = 'genome'
READS_DIR = 'reads'
ALIGNMENTS_DIR = 'alignments'

rule all:
    input:
        f"{BASE_DIR}/index/yeast.1.bt2",
        expand(join(BASE_DIR, ALIGNMENTS_DIR, '{sample}.sam'), sample=SAMPLES)

rule index: 
    input: 
        genome = f'{BASE_DIR}/{GENOME_DIR}/genome.fasta'
    output:
        index = f'{BASE_DIR}/index/yeast.1.bt2'
    params:
        index_name = 'yeast',
        index_folder = f'{BASE_DIR}/index'
    threads:
        8
    message : "Building index"
    shell:
        "bowtie2-build --threads {threads} -f {input.genome} {params.index_name}"


rule align:
    input:
        index = f'{BASE_DIR}/index/yeast.1.bt2',
        reads = join(BASE_DIR, READS_DIR, '{sample}_subsamp.fastq.gz'),
    output:
        sam = join(BASE_DIR, ALIGNMENTS_DIR, '{sample}.sam')
    params:
        index_name = f"{BASE_DIR}/index/yeast",
    threads :
        8
    message: "Aligning reads"
    shell:
        "bowtie2 -x {params.index_name} -U {input.reads} -S {output.sam}"